lib/active_job_status/job_tracker.rb

```diff
@@ -24,18 +25,28 @@ def performing
     end
 
     def completed
+      previous_status = store.fetch(job_id)
       store.write(
         job_id,
         JobStatus::COMPLETED.to_s
       )
+      remove_from_batch if batch_id && previous_status && previous_status != JobStatus::COMPLETED.to_s
     end
 
     def deleted
+      previous_status = store.fetch(job_id)
       store.delete(job_id)
+      remove_from_batch if batch_id && previous_status && previous_status != JobStatus::COMPLETED.to_s
```

Comment:
is it better to define remove_from_batch like this?

```ruby
def remove_from_batch(previous_status)
  if batch_id && previous_status && previous_status != JobStatus::COMPLETED.to_s
    # do stuff in existing remove_from_batch
  end
end
```

this helps DRY up the two repeated conditions